src/terrama2/core/interpreter/PythonInterpreter.cpp

```diff
@@ -172,6 +179,27 @@ void terrama2::core::PythonInterpreter::runScript(const std::string& script)
   }
 }
 
+std::string terrama2::core::PythonInterpreter::runScriptWithStringResult(const std::string& script, const std::string& variableToReturn)
+{
+  auto lock = holdState();
+
+  using namespace boost::python;
+
+  try
+  {
+    PyObject *main = PyImport_AddModule("__main__");
+    PyRun_SimpleString(script.c_str());
+    PyObject *catcher = PyObject_GetAttrString(main, variableToReturn.c_str());
+    std::string out = PyString_AsString(catcher);
+
+    return out;
+  }
+  catch( error_already_set )
```

Comment:
Prefer to catch an exception using `const` and `&` respectively to help compiler optimizations (Even if it is not used). 

**Note** Using a copy like that, you may face the problematic inheritance hierarchies **if you need to use** the caught object.